File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n 
    
    def dfs(start):
        visited[start] = True
        
        for i in range(n):
            # 실수한 점-> 무엇을 비교하는지 (visited의 인덱스가 무엇이 되어야 하는 지 정확히 파악하기)
            if computers[start][i] and not visited[i]:
                dfs(i)
    
    for i in range(n):
        if visited[i]:
            continue
        dfs(i)
        answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
public class Solution {
	public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
        	if (visited[i]) continue;
        	
        	dfs(computers, n, visited, i);
        	answer++;
        }
        
        
        
        return answer;
    }
	
	private void dfs(int[][] computers, int n, boolean[] visited, int start) {
		visited[start] = true;
		
		for (int i = 0; i < n; i++) {
			if (computers[start][i] !=  0 && !visited[i]) dfs(computers, n, visited, i);
		}
	};
}
